app/models/transaction.rb

```diff
@@ -0,0 +1,60 @@
+class Transaction < ApplicationRecord
+  #
+  ## Associations
+  #
+  belongs_to :account
+
+  #
+  ## Validations
+  #
+  validates(
+    :description,
+    presence: true,
+    length: {
+      maximum: 900
+    }
+  )
+  validates(
+    :amount,
+    presence: true
+  )
+  validates(
+    :transaction_type,
+    presence: true
+  )
+
+  #
+  ## Callbacks
+  #
+  before_validation :check_account_number, :check_balance
+  after_create :remove_amount_from_balance
+
+  def check_account_number
+    if self.transaction_type == "NEFT"
+      @account = Account.find_by(account_number: self.account_number)
+      if !@account.present?
```

Comment:
@babarajay , How would you do this using positive condition?

Reply:
I'll remove this condition and will write new positive condition.